pkg/sampling/internal/bytes/randomness.go

```diff
@@ -0,0 +1,36 @@
+package bytes
+
+import (
+	"encoding/hex"
+	"errors"
+
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+// ErrRValueSize is returned for r-values != NumHexDigits hex digits.
+var ErrRValueSize = errors.New("r-value must have 14 hex digits")
+
+// Randomness may be derived from r-value or TraceID.
+type Randomness struct {
+	// bytes[0] is unused, so that the relevant portion of these 8
+	// bytes align with the TraceID's second 8 bytes.
```

Comment:
What is the benefit of aligning the indices? It looks like you use array slices for any comparisons.